crates/ingress-rpc/src/service.rs

```diff
@@ -43,6 +72,59 @@ impl<Queue> IngressService<Queue> {
             queue,
         }
     }
+
+    async fn validate_tx(&self, envelope: &OpTxEnvelope) -> Result<()> {
+        let sender = envelope.recover_signer().unwrap_or_default();
+        let account = self.provider.fetch_account_info(sender).await?;
+
+        // skip eip4844 transactions
+        if envelope.is_eip4844() {
+            return Err(anyhow::anyhow!("EIP-4844 transactions are not supported"));
+        }
+
+        // from: https://github.com/paradigmxyz/reth/blob/3b0d98f3464b504d96154b787a860b2488a61b3e/crates/optimism/txpool/src/supervisor/client.rs#L76-L84
+        // it returns `None` if a tx is not cross chain, which is when `inbox_entries` is empty in the snippet above.
+        // we can do something similar where if the inbox_entries is non-empty then it is a cross chain tx and it's something we don't support
+        if let Some(access_list) = envelope.access_list() {
+            let inbox_entries = access_list
+                .iter()
+                .filter(|entry| entry.address == CROSS_L2_INBOX_ADDRESS);
+            if inbox_entries.count() > 0 {
+                return Err(anyhow::anyhow!("Interop transactions are not supported"));
+            }
+        }
+
+        // error if account is 7702 but tx is not 7702
+        if account.code_hash != KECCAK_EMPTY && !envelope.is_eip7702() {
+            return Err(anyhow::anyhow!(
+                "Account is a 7702 account but transaction is not EIP-7702"
+            ));
+        }
+
+        // error if tx nonce is not the latest
+        // https://github.com/paradigmxyz/reth/blob/a047a055ab996f85a399f5cfb2fe15e350356546/crates/transaction-pool/src/validate/eth.rs#L611
+        if envelope.nonce() < account.nonce {
+            return Err(anyhow::anyhow!("Nonce is not the latest"));
+        }
+
+        // error if execution cost costs more than balance
+        if envelope.value() > account.balance {
```

Comment:
i'm not sure if `value()` is correct here.

it's difficult to get the `cost()` from an `OpPooledTransaction` or Op envelope and I couldn't figure it out

Reply:
I suspect this is some combo of gasPrice * gasLimit + value. We should try and import and if not we can reimplement